File: src/main/java/com/ort/fw/util/WolfMansionUserInfoUtil.java
package com.ort.fw.util;

import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;

import com.ort.fw.security.UserInfo;

public class WolfMansionUserInfoUtil {

    private WolfMansionUserInfoUtil() {
    }

    public static UserInfo getUserInfo() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        if (authentication == null) {
            return null;
        }
        if (authentication.getPrincipal() instanceof UserInfo) {
            return UserInfo.class.cast(authentication.getPrincipal());
        }
        return null;
    }
}

File: backend/src/main/kotlin/com/ort/app/fw/util/WolfMansionUserInfoUtil.kt
package com.ort.app.fw.util

import com.ort.app.fw.security.JwtTokenService
import com.ort.app.fw.security.UserInfo
import com.ort.dbflute.allcommon.CDef
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.oauth2.jwt.Jwt

object WolfMansionUserInfoUtil {

    fun getUserInfo(): UserInfo? {
        val authentication = SecurityContextHolder.getContext().authentication ?: return null
        val principal = authentication.principal
        return when (principal) {
            is UserInfo -> principal
            is Jwt -> fromJwt(principal)
            else -> null
        }
    }

    private fun fromJwt(jwt: Jwt): UserInfo? {
        val userName = jwt.subject ?: return null
        val authorityCode = jwt.getClaimAsString(JwtTokenService.CLAIM_AUTHORITY)
        val authority = authorityCode?.let { CDef.Authority.codeOf(it) } ?: CDef.Authority.プレイヤー
        return UserInfo().apply {
            setUsername(userName)
            setPassword("") // JWT 認証では DB のパスワードは不要
            setAuthority(authority)
        }
    }
}
